pkg/kubelet/certificate/bootstrap/bootstrap.go

```diff
@@ -98,23 +140,64 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
 		}
 	}
 
-	if err := waitForServer(*bootstrapClientConfig, 1*time.Minute); err != nil {
-		klog.Warningf("Error waiting for apiserver to come up: %v", err)
+	interval := wait.Backoff{
+		Duration: time.Second,
+		Factor:   1.1,
+		Steps:    20,
+		Jitter:   0.2,
+		Cap:      30 * time.Second,
 	}
 
-	certData, err := csr.RequestNodeCertificate(bootstrapClient.CertificateSigningRequests(), keyData, nodeName)
+	// get the correct transport from our config
+	transportCfg, err := bootstrapClientConfig.TransportConfig()
 	if err != nil {
 		return err
 	}
-	if _, err := store.Update(certData, keyData); err != nil {
+	tlsConfig, err := transport.TLSConfigFor(transportCfg)
+	if err != nil {
 		return err
 	}
-	if err := os.Remove(privKeyPath); err != nil && !os.IsNotExist(err) {
-		klog.V(2).Infof("failed cleaning up private key file %q: %v", privKeyPath, err)
+	timeoutClientConfig := bootstrapClientConfig.DeepCopy()
+	timeoutClientConfig.TLSClientConfig = transport.TLSClientConfig{}
+
+	for interval.Steps > 0 {
+		// use a custom transport that disables keep alives
+		t := interval.Step()
+		timeoutClientConfig.Transport = &http.Transport{
+			TLSClientConfig:   tlsConfig,
+			DisableKeepAlives: true,
+			DialContext:       (&net.Dialer{Timeout: t}).DialContext,
+		}
+
+		bootstrapClient, err := certificates.NewForConfig(timeoutClientConfig)
+		if err != nil {
+			if t, ok := err.(timeoutError); ok && t.Timeout() {
```

Comment:
Why would creating a new certs client timeout?

Reply:
This is the first api call made to join a node to a cluster, it races with network programming on GCE, GCE has bugs where ACKs disappear into the ether.  See the referenced PR

Reply:
But isn't `csr.RequestNodeCertificate` below the API call? Why would `certificates.NewForConfig` result in a request?

Reply:
Moved out of this PR so as not to block this if we can't get review time by tomorrow.